kpops/components/base_components/models/from_section.py

```diff
@@ -16,26 +16,53 @@ class InputTopicTypes(str, Enum):
 
     INPUT = "input"
     EXTRA = "extra"
+    PATTERN = "pattern"
     INPUT_PATTERN = "input-pattern"
     EXTRA_PATTERN = "extra-pattern"
 
 
 class FromTopic(BaseModel):
     """Input topic
 
-    :param type: Topic type
-    :type type: InputTopicTypes
-    :param role: Custom identifier belonging to a topic, provide only if `type` is `extra` or `extra-pattern`
-    :type role: str | None
+    :param type: Topic type, defaults to None
+    :type type: InputTopicTypes | None, optional
+    :param role: Custom identifier belonging to a topic, provide only if `type` is `extra` or `extra-pattern`.
+        When `role` is not `None`, `type: pattern` is equal to `type: extra-pattern`, defaults to None
+    :type role: str | None, optional
     """
 
-    type: InputTopicTypes = Field(..., description=describe_attr("type", __doc__))
+    type: InputTopicTypes | None = Field(
+        default=None, description=describe_attr("type", __doc__)
+    )
     role: str | None = Field(default=None, description=describe_attr("role", __doc__))
 
     class Config(DescConfig):
         extra = Extra.forbid
         use_enum_values = True
 
+    def __init__(self, **kwargs):
+        kwargs["type"] = self.__assign_type(kwargs)
+        super().__init__(**kwargs)
+
+    @staticmethod
+    def __assign_type(kwargs: dict) -> str:
```

Comment:
also extract the kwargs
```suggestion
    def __assign_type(type: InputTopicTypes | None, role: str | None) -> InputTopicTypes:
```